solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java

```diff
@@ -988,8 +987,8 @@ protected NamedList<Object> sendRequest(SolrRequest<?> request, List<String> inp
 
     boolean sendToLeaders = false;
 
-    if (request instanceof IsUpdateRequest) {
-      sendToLeaders = ((IsUpdateRequest) request).isSendToLeaders() && this.isUpdatesToLeaders();
+    if (request.getRequestType() == SolrRequestType.UPDATE) {
+      sendToLeaders = request.shouldSendToLeaders() && this.isUpdatesToLeaders();
```

Comment:
I know casting is never elegant but shouldSendToLeaders() doesn't belong on SolrRequest base class